xterm/index.cgi

```diff
@@ -7,13 +7,72 @@
 # XXX Virtualmin integration?
 
 require './xterm-lib.pl';
+&ReadParse();
 
+# Get Webmin current version for links serial
+my $wver = &get_webmin_version();
+$wver =~ s/\.//;
+
+# Build Xterm dependency links
+my $termlinks = 
+	{ 'css' => ['xterm.css?$wver'],
+	  'js'  => ['xterm.js?$wver',
+	  'xterm-addon-attach.js?$wver',
+	  'xterm-addon-fit.js?$wver'] };
+
+# Print header
 &ui_print_header(undef, $text{'index_title'}, "", undef, 1, 1, 0, undef,
-		 "<link rel=stylesheet href=xterm.css>\n".
-		 "<script src=xterm.js></script>\n".
-		 "<script src=xterm-addon-attach.js></script>\n"
+		 "<link rel=stylesheet href=\"$termlinks->{'css'}[0]\">\n".
+		 "<script src=\"$termlinks->{'js'}[0]\"></script>\n".
+		 "<script src=\"$termlinks->{'js'}[1]\"></script>\n".
+		 "<script src=\"$termlinks->{'js'}[2]\"></script>\n"
 		);
 
+# Set column size depending on the browser window size
+my $screen_width = int($ENV{'HTTP_X_AGENT_WIDTH'}) || int($in{'w'});
+my $screen_height = int($ENV{'HTTP_X_AGENT_HEIGHT'}) || int($in{'h'});
+if (!$screen_width ||
+    !$screen_height) {
+	print "<script>location.href = location.pathname + '?w=' + window.innerWidth + '&h=' + window.innerHeight;</script>";
```

Comment:
What's the purpose of these w= and h= URL parameters?

Reply:
These return browser's current window width and height.

Reply:
Will this case the page to be reloaded though? And do we really need this, or can we use width=100% ?

Reply:
Yes, it will but only if headers (with width and height) aren't sent, i.e. in old themes.

We need absolute values, not relative, so `%` won't work.

Reply:
Why not use the `redirect` sub instead of setting `location.href` ?

Reply:
> Why not use the `redirect` sub instead of setting `location.href` ?

The trick here is to get container size, which depends on the user screen/resolution/scaling. Authentic Theme won't need it, as container size will be sent with headers. Older themes will have to do quick reload (it's un-noticeable though and URL will be cleared afterward to support page resize).